pattern-editor/src/App.jsx

```diff
@@ -101,6 +102,37 @@ export default function App() {
     pushPattern(newPattern);
   }
 
+  // the idea is that you want to make the pixel in that view that colour
+  // eventually we should use rules to make this modification 
+  // but for now we just apply the settings chosen in OptionEditor
+  function updateForDirection(direction) {
```

Comment:
sort of personal preference, but instead of having this as a nested function, could you just have `updateForDirection(viewRow, viewCol, direction) {...}` and then what you pass as the `updatePixel` prop would be `(row, col) => updateForDirection(row, col, DIRECTION.<insert direction here>)` ? Might be a little more readable

Reply:
that would definitely work just as well, I think I like currying because a) I'm contractually obliged b) it kind of matches the style of the pattern editor (no anonymous functions)